src/core_plugins/tagcloud/public/tag_cloud.js

```diff
@@ -250,36 +264,34 @@ class TagCloud extends EventEmitter {
       return;
     }
 
-    clearTimeout(this._handle);
-    this._handle = setTimeout(() => {
-      this._handle = null;
-      this._updateContainerSize();
-      if (keepLayout && this._currentJob && this._queue.length === 0) {
-        this._scheduleLayout({
-          words: this._currentJob.words.map(tag => {
-            return {
-              x: tag.x,
-              y: tag.y,
-              rotate: tag.rotate,
-              size: tag.size,
-              text: tag.text
-            };
-          })
-        });
-      } else {
-        this._updateLayout();
-      }
-    }, 0);//unhook from callstack. this avoids kicking off multiple layouts if multiple changes come in succession
+    this._updateContainerSize();
+    if (keepLayout && this._currentJob && !this._pendingJob) {
+      //no need to recompute the layout
+      this._scheduleLayout({
+        words: this._currentJob.words.map(tag => {
+          return {
+            refreshLayout: false,
```

Comment:
I don't think i like the refreshLayout variable being passed around. Before you were directly calling the _updateLayout here which was easier to understand for me. Flow something like invalidate -> if we need to recalc layout updateLayout -> then all the other steps were always the same. I would make updateLayout return a promise an actually call _scheduleLayout from this function, so its clear that updateLayout is just one extra step that might be needed in some scenarios.